src/Mvc/Mvc.Formatters.Xml/src/XmlDataContractSerializerInputFormatter.cs

```diff
@@ -120,7 +120,17 @@ public override async Task<InputFormatterResult> ReadRequestBodyAsync(InputForma
             Stream readStream = new NonDisposableStream(request.Body);
             var disposeReadStream = false;
 
-            if (!request.Body.CanSeek && !_options.SuppressInputFormatterBuffering)
+            if (readStream.CanSeek)
+            {
+                // The most common way of getting here is the user has request buffering on.
+                // However, request buffering isn't eager, and consequently it will peform pass-thru synchronous
+                // reads as part of the deserialization.
+                // To avoid this, drain and reset the stream.
+                var position = request.Body.Position;
```

Comment:
Do you have a test where this is non-zero?

Reply:
Done